src/app/products/floating-rates/create-floating-rate/create-floating-rate.component.ts

```diff
@@ -0,0 +1,116 @@
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { ProductsService } from '../../products.service';
+
+/**
+ * Create Floating Rate Component.
+ */
+@Component({
+  selector: 'mifosx-create-floating-rate',
+  templateUrl: './create-floating-rate.component.html',
+  styleUrls: ['./create-floating-rate.component.scss']
+})
+export class CreateFloatingRateComponent implements OnInit {
+
+  /** Minimum floating rate period date allowed. */
+  minDate = new Date();
+  /** Floating Rate Form. */
+  floatingRateForm: FormGroup;
+
+  /**
+   * @param {Router} router Router for navigation.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Product Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {DatePipe} datePipe Date Pipe.
+   */
+  constructor(private router: Router,
+              private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private datePipe: DatePipe) { }
+
+  /**
+   * Creates the Floating Rate Form.
+   */
+  ngOnInit() {
+    this.createFloatingRateForm();
+  }
+
+  /**
+   * Creates the Floating Rate Form.
+   */
+  createFloatingRateForm() {
+    this.floatingRateForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'isBaseLendingRate': [false],
+      'isActive': [false],
+      'ratePeriods': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Creates the Floating Rate Periods Form.
+   * @returns {FormGroup} Floating Rate Period Form.
+   */
+  createFloatingRatePeriodsForm(): FormGroup {
+    return this.formBuilder.group({
+      'fromDate': ['', Validators.required],
+      'interestRate': ['', Validators.required],
+      'isDifferentialToBaseLendingRate': [false]
+    });
+  }
+
+  /**
+   * Gets the floating rate periods form array.
+   * @returns {FormArray} Floating rate periods form array.
+   */
+  get ratePeriods(): FormArray {
+    return this.floatingRateForm.get('ratePeriods') as FormArray;
+  }
+
+  /**
+   * Removes the floating rate period form row from given floating rate period form at given index.
+   * @param {FormArray} ratePeriods Floating Rate Periods Form Array.
+   * @param {number} index Array index from where floating rate period form needs to be removed.
+   */
+  removeFloatingRatePeriod(ratePeriods: FormArray, index: number) {
+    ratePeriods.removeAt(index);
+  }
+
+  /**
+   * Adds a row in the floating rate period form.
+   */
+  addFloatingRatePeriod() {
+    this.ratePeriods.push(this.createFloatingRatePeriodsForm());
+  }
+
+  /**
+   * Submits the floating rate form and creates floating rate,
+   * if successful redirects to view created floating rate.
+   */
+  submit() {
+    const dateFormat = 'yyyy-MM-dd';
+    const floatingRate = this.floatingRateForm.value;
+    // TODO: Update once language and date settings are setup
+    if (floatingRate.ratePeriods) {
```

Comment:
The `if..else` statement here is mainly used to set `floatingRate.ratePeriods` undefined so that it doesn't send an empty array (which will eventually lead to an error).